packages/financial-templates-lib/src/clients/InsuredBridgeL2Client.ts

```diff
@@ -122,14 +127,85 @@ export class InsuredBridgeL2Client {
     });
   }
 
-  async getFundsDepositedEvents(blockSearchConfig: { fromBlock: number; toBlock: number }): Promise<EventData[]> {
-    return await this.bridgeDepositBox.getPastEvents("FundsDeposited", blockSearchConfig);
+  async getFundsDepositedEvents(blockSearchConfig: BlockSearchConfig): Promise<EventData[]> {
+    const eventsData = await this.getEventsForMultipleProviders(
+      this.fallbackL2Web3s,
+      this.bridgeDepositBoxContract,
+      "FundsDeposited",
+      blockSearchConfig
+    );
+    if (!eventsData.success)
+      throw new Error(
+        `FundsDeposited transaction hash ${eventsData.missingEvent} found in fallback l2 provider not found in first l2 web3 provider`
+      );
+    return eventsData.events[0];
   }
 
-  async getWhitelistTokenEvents(blockSearchConfig: { fromBlock: number; toBlock: number }): Promise<EventData[]> {
-    return await this.bridgeDepositBox.getPastEvents("WhitelistToken", blockSearchConfig);
+  async getWhitelistTokenEvents(blockSearchConfig: BlockSearchConfig): Promise<EventData[]> {
+    const eventsData = await this.getEventsForMultipleProviders(
+      this.fallbackL2Web3s,
+      this.bridgeDepositBoxContract,
+      "WhitelistToken",
+      blockSearchConfig
+    );
+    if (!eventsData.success)
+      throw new Error(
+        `WhitelistToken transaction hash ${eventsData.missingEvent} found in fallback l2 provider not found in first l2 web3 provider`
+      );
+    return eventsData.events[0];
   }
 
+  /**
+   * Fetches specified event data for all input web3 providers. Assumes that the first fetched EventData array contains
+   * the "control" events. Compares the "control" events against each of the other EventData arrays. Returns false
+   * if any of the "control" events are NOT found exactly in ALL of the other EventData arrays.
+   * @param web3s Web3 providers to check for target event.
+   * @param contract Contract to query target event on.
+   * @param eventName Name of target event
+   * @param blockSearchConfig Target event search config
+   * @returns Object containing success of event query, missing event if not found in all providers, and all event data
+   */
+  async getEventsForMultipleProviders(
```

Comment:
@mrice32 here's the module we can move into some external utilities file. I haven't moved it yet bc i'm not sure where to place it. Thoughts?

Reply:
@daywiss do you have thoughts? I could see us putting this in the SDK, common, or financial-templates-lib/src/utils.

Reply:
I'll leave it in this package for now, but probably move to a new file within `src/helpers` instead of the L2Client. And if we want to use it elsewhere in the future we can move it

Reply:
ye I I dont think this should go in the SDK. perhaps `common` in a new util called `EventUtils` or something?

Reply:
hmmm I like that better actually